### PR TITLE
Initial arm64e support

### PR DESCRIPTION
**Pull request by vargaz**
- Fix the c++ build.
- Remove some unused prototypes.
- Initial arm64e support.

### CODE REVIEW

File: mono/mini/tramp-arm64.c [hunk 1]
```diff
@@ -336,7 +337,8 @@ mono_arch_create_specific_trampoline (gpointer arg1, MonoTrampolineType tramp_ty
 	code = mono_arm_emit_imm64 (code, ARMREG_IP1, (guint64)arg1);
 	code = mono_arm_emit_imm64 (code, ARMREG_IP0, (guint64)tramp);
 
-	arm_brx (code, ARMREG_IP0);
+	code = mono_arm_emit_brx (code, ARMREG_IP0);
+	//arm_brx (code, ARMREG_IP0);
```
Comment (akoeplinger) [inline mono/mini/tramp-arm64.c:341]: dead commented code

File: mono/utils/Makefile.am [hunk 2]
```diff
@@ -236,7 +236,8 @@ monoutils_sources = \
 	ward.h \
 	options.h \
 	options-def.h \
-	options.c
+	options.c \
+	ftnptr.h
```
Comment (akoeplinger) [inline mono/utils/Makefile.am:240]: this file will be deleted by #47758 so probably best to not touch it

File: mono/mini/mini-runtime.c [hunk 3]
```diff
@@ -3433,6 +3435,7 @@ mono_jit_runtime_invoke (MonoMethod *method, void *obj, void **params, MonoObjec
 		if (!is_ok (error))
 			return NULL;
 	} else {
+		//runtime_invoke = (MonoObject *(*)(MonoObject *, void **, MonoObject **, void *))MINI_ADDR_TO_FTNPTR (info->runtime_invoke);
```
Comment (akoeplinger) [inline mono/mini/mini-runtime.c:3438]: leftover debugging code? or TODO?

File: mono/mini/mini-arm64.c [hunk 4]
```diff
@@ -5434,7 +5484,8 @@ mono_arch_build_imt_trampoline (MonoVTable *vtable, MonoDomain *domain, MonoIMTC
 
 	MINI_END_CODEGEN (buf, code - buf, MONO_PROFILER_CODE_BUFFER_IMT_TRAMPOLINE, NULL);
 
-	return buf;
+	return MINI_ADDR_TO_FTNPTR (buf);
+	//return buf;
```
Comment (akoeplinger) [inline mono/mini/mini-arm64.c:5488]: dead commented code

File: mono/utils/Makefile.am (hunk 2 above)
Comment (vargaz) [reply mono/utils/Makefile.am:240]: Its still present in this repo.

Comment (akoeplinger) [reply mono/utils/Makefile.am:240]: ah, didn't see that this is mono/mono .... :smile:

**Comment by vargaz** [comment]
@monojenkins build failed

**Review by imhameed — APPROVED** [review]

**Comment by akoeplinger** [comment]
Ah, the /azp command only works _before_ a PR is merged.

@vargaz by the way I noticed this causes a build failure on the [iOS SDK job](https://jenkins.mono-project.com/job/archive-mono/job/master/3020/):

```
[2021-02-04T03:24:09.120Z]   CCLD     libmonosgen-2.0.la
[2021-02-04T03:24:09.120Z] Undefined symbols for architecture arm64:
[2021-02-04T03:24:09.120Z]   "_mono_arm_emit_blrx", referenced from:
[2021-02-04T03:24:09.120Z]       _mono_arch_set_breakpoint in libmini.a(libmini_la-mini-arm64.o)
[2021-02-04T03:24:09.120Z]   "_mono_arm_emit_brx", referenced from:
[2021-02-04T03:24:09.120Z]       _get_delegate_invoke_impl in libmini.a(libmini_la-mini-arm64.o)
[2021-02-04T03:24:09.120Z]       _mono_arch_get_gsharedvt_arg_trampoline in libmini.a(libmini_la-tramp-arm64-gsharedvt.o)
[2021-02-04T03:24:09.120Z] ld: symbol(s) not found for architecture arm64
[2021-02-04T03:24:09.120Z] clang: error: linker command failed with exit code 1 (use -v to see invocation)
[2021-02-04T03:24:09.120Z] gmake[6]: *** [Makefile:1752: libmonosgen-2.0.la] Error 1
```

**Pull request merged**